Lib/asyncio/base_events.py

```diff
@@ -83,18 +83,18 @@ def _set_reuseport(sock):
                              'SO_REUSEPORT defined but not implemented.')
 
 
-def _is_stream_socket(sock):
+def _is_stream_socket(sock_type):
     # Linux's socket.type is a bitmask that can include extra info
     # about socket, therefore we can't do simple
     # `sock_type == socket.SOCK_STREAM`.
-    return (sock.type & socket.SOCK_STREAM) == socket.SOCK_STREAM
+    return (sock_type & socket.SOCK_STREAM) == socket.SOCK_STREAM
```

Comment:
`socket.SOCK_STREAM` is not a bit, it's a value, but `socket.SOCK_NONBLOCK` is a bit. That's why it is weird and complicated. Values and socket creation flags are mixed. For example, this check will fail for `sock_type == socket.SOCK_SEQPACKET`:
```python
>>> (socket.SOCK_SEQPACKET & socket.SOCK_STREAM) == socket.SOCK_STREAM
True
```
All possible values on Linux:
```python
>>> list(socket.SocketKind)
[<SocketKind.SOCK_STREAM: 1>,
 <SocketKind.SOCK_DGRAM: 2>,
 <SocketKind.SOCK_RAW: 3>,
 <SocketKind.SOCK_RDM: 4>,
 <SocketKind.SOCK_SEQPACKET: 5>,
 <SocketKind.SOCK_NONBLOCK: 2048>,
 <SocketKind.SOCK_CLOEXEC: 524288>]
```